src/libsync/abstractnetworkjob.cpp

```diff
@@ -315,4 +332,48 @@ QString errorMessage(const QString& baseError, const QByteArray& body)
     return msg;
 }
 
+QByteArray requestVerb(const QNetworkReply& reply)
+{
+    switch (reply.operation()) {
+    case QNetworkAccessManager::HeadOperation:
+        return "HEAD";
+    case QNetworkAccessManager::GetOperation:
+        return "GET";
+    case QNetworkAccessManager::PutOperation:
+        return "PUT";
+    case QNetworkAccessManager::PostOperation:
+        return "POST";
+    case QNetworkAccessManager::DeleteOperation:
+        return "DELETE";
+    case QNetworkAccessManager::CustomOperation:
+        return reply.request().attribute(QNetworkRequest::CustomVerbAttribute).toByteArray();
+    case QNetworkAccessManager::UnknownOperation:
+        break;
+    }
+    return QByteArray();
+}
+
+QString networkReplyErrorString(const QNetworkReply& reply)
+{
+    QString base = reply.errorString();
+    int httpStatus = reply.attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    QString httpReason = reply.attribute(QNetworkRequest::HttpReasonPhraseAttribute).toString();
+
+    // Only adjust HTTP error messages of the expected format.
+    if (httpReason.isEmpty() || httpStatus == 0 || !base.contains(httpReason)) {
+        return base;
+    }
+
```

Comment:
@guruz  This commit will continue to work (i.e. replace this error message) even if the Qt error message changes in the way described in https://code.qt.io/cgit/qt/qtbase.git/commit/?id=4efa50a5e37497bc3a3dd30f27db7d8100e3f6a5. I would merge it.